core/ledger/ledger_interface.go

```diff
@@ -624,10 +624,25 @@ func (e *InvalidCollNameError) Error() string {
 // PvtdataHashMismatch is used when the hash of private write-set
 // does not match the corresponding hash present in the block
 // See function `PeerLedger.CommitPvtData` for the usages
+// Only one of the fields ExpectedHash or BootKVsMismatch will be present
+// ExpectedHash will be present if the BlockNum is greater than the last
+// block number present in the bootstrapping snapshot, if the ledger is
+// bootstrapped from a snapshot
```

Comment:
Looks like neither `ExpectedHash` nor `BooKVsMismatch` is used in the `reconciler.go` (not even in the warning message). As it is kind of dead-code or not being used in the production code, I would prefer these two fields to be removed. Otherwise, we can print them in the warning message but not sure what help it would provide.

Reply:
yes, I'll better remove this altogether.